### PR TITLE
Fixes in the PMIx test suite

### PR DESCRIPTION
**Pull request by artpol84**
* Fix error code propagation
* Fix "success" report for failing cases
* Minor improvements with debugging output

### CODE REVIEW

**Comment by artpol84** [comment]
@rhc54 This should fix some of the issues you observed with the test suite.
Please, give it a try and let me know.

**NOTE**: that this fix uncovers a bug with PMIx fence not being called. I tracked it down and fixed. Will PR shortly.

**Comment by artpol84** [comment]
The 2 failures are expected as these fixes expose the problem with Fence fixed in https://github.com/pmix/pmix/pull/1235.

Thus the right sequence would be to merge #1235 and reevaluate this one.

File: test/test_server.c [hunk 1]
```diff
@@ -841,15 +843,11 @@ int server_finalize(test_params *params)
...
         /* wait for all servers are finished */
-        ret = srv_wait_all(10.0);
-        if (!pmix_list_is_empty(server_list)) {
-            total_ret += ret;
-        }
+        total_ret += srv_wait_all(10.0);
         PMIX_LIST_RELEASE(server_list);
         TEST_VERBOSE(("SERVER %d FINALIZE PID:%d with status %d",
-                    my_server_id, getpid(), ret));
+	                    my_server_id, getpid(), total_ret));
```
Comment (artpol84) [inline test/test_server.c]:
```suggestion
                        my_server_id, getpid(), total_ret));
```

**Comment by artpol84** [comment]
bot:retest

**Comment by ibm-ompi** [comment]
The IBM CI (Cross-version) build failed! Please review the log, linked below.

 Gist: https://gist.github.com/9453ee085ad9220e78018a8b2878ad68

**Comment by ibm-ompi** [comment]
The IBM CI (GNU Compiler) build failed! Please review the log, linked below.

 Gist: https://gist.github.com/be002a1395a2256672363b0d1d955930

**Comment by ibm-ompi** [comment]
The IBM CI (XL Compiler) build failed! Please review the log, linked below.

 Gist: https://gist.github.com/236a92c185825066e7336f9bcf5b4d16

**Comment by ibm-ompi** [comment]
The IBM CI (Cross-version) build failed! Please review the log, linked below.

 Gist: https://gist.github.com/d1cc66d2692aea8c4417e737c13870ae

**Comment by artpol84** [comment]
bot:ibm:retest

**Comment by artpol84** [comment]
bot:retest

**Comment by artpol84** [comment]
bot:retest

**Comment by jjhursey** [comment]
Ok. I think this one is ready for review. Right?

**Comment by artpol84** [comment]
I'm working on one issue now that I'm seeing with the test suite. I am sure it is not related, but if you want we can wait for me to finish.

**Comment by jjhursey** [comment]
Per teleconf:
 * @artpol84 seeing occasional hang in testing. May not be a problem in the test suite, but a race in the library that needs to be investigated further.
 * Suggested that maybe we have a CI test that runs `make check` multiple times to catch the run-once-ok, run-twice-fail scenario.

**Comment by artpol84** [comment]
@jjhursey
I think I tracked down the issue and I think it’s a blocker for 3.1

I’ll open the pr later today.
Please wait for it before rolling out rc2

**Comment by artpol84** [comment]
I think this one is good to go. It does introduce a bunch of useful stuff.

**Pull request merged**